### PR TITLE
:sparkles:  move provider interface into the manager

### PR DESCRIPTION
**Pull request by ezrasilvera**
## Summary
Moving the provider interface and all needed definitions out of the "Provider folder"
In the future provider implementation will be in different repositories 
## Related issue(s)

Fixes #

### CODE REVIEW

**Comment by effi-ofer** [comment]
This is a step in the right direction.  LGTM.

File: pkg/clustermanager/reconcile_logicalcluster.go [hunk 1]
```diff
@@ -22,9 +22,9 @@ import (
 	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/klog/v2"
 
-	clusterproviderclient "github.com/kubestellar/kubestellar/cluster-provider-client"
-	cluster "github.com/kubestellar/kubestellar/cluster-provider-client/cluster"
 	lcv1alpha1 "github.com/kubestellar/kubestellar/pkg/apis/logicalcluster/v1alpha1"
+	cluster "github.com/kubestellar/kubestellar/pkg/clustermanager/providerclient"
```
Comment (andreyod) [inline pkg/clustermanager/reconcile_logicalcluster.go:26]: Name this import `clusterprovider` to be consistent.

File: pkg/clustermanager/reconcile_logicalcluster.go [hunk 2]
```diff
@@ -22,9 +22,9 @@ import (
...
+	// clusterproviderclient "github.com/kubestellar/kubestellar/pkg/clustermanager/provider-client-interface"
```
Comment (andreyod) [inline pkg/clustermanager/reconcile_logicalcluster.go:27]: Please remove this comment.

File: pkg/clustermanager/discovery.go [hunk 3]
```diff
@@ -24,14 +24,16 @@ import (
 	"k8s.io/apimachinery/pkg/watch"
 	"k8s.io/klog/v2"
 
-	clusterproviderclient "github.com/kubestellar/kubestellar/cluster-provider-client"
-	clusterprovider "github.com/kubestellar/kubestellar/cluster-provider-client/cluster"
 	lcv1alpha1apis "github.com/kubestellar/kubestellar/pkg/apis/logicalcluster/v1alpha1"
 	edgeclient "github.com/kubestellar/kubestellar/pkg/client/clientset/versioned"
+
+	//	clusterproviderclient "github.com/kubestellar/kubestellar/pkg/clustermanager/provider-client-interface"
+	//	clusterproviderclient "github.com/kubestellar/kubestellar/pkg/clustermanager/provider-client-interface"
```
[the remark this answers is not in the document]
Comment (ezrasilvera) [reply pkg/clustermanager/discovery.go]: done

File: pkg/clustermanager/reconcile_logicalcluster.go (hunk 1 above)
Comment (ezrasilvera) [reply pkg/clustermanager/reconcile_logicalcluster.go]: done

**Pull request merged**